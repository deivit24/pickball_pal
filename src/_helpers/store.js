import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
