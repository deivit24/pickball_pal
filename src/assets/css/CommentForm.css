#CommentForm button {
  background-color: white;
  color: #3f50b5;
  font-weight: 700;
  margin: 10px auto;
}
#CommentForm button:hover {
  background-color: #3f50b5;
  color: white;
  border: 1px solid white;
  font-weight: 700;
  margin: 10px auto;
}

#CommentForm label {
  color: black;
}
#CommentForm .black {
  color: black;
}

#CommentForm .white {
  color: white;
}

#CommentForm .Mui-focused::before {
  border-bottom-color: white;
}

#CommentForm .Mui-focused::after {
  border-bottom-color: white;
}
