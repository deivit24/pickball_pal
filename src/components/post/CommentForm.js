import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { addComment } from '../../_actions/post';

import Input from '@material-ui/core/Input';
import Button from '@material-ui/core/Button';

const CommentForm = ({ postId }) => {
  const dispatch = useDispatch();
  const handleChange = (e) => {
    setText(e.target.value);
  };
  const [text, setText] = useState('');
  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(addComment(postId, { text }));
    setText('');
  };
  return (
    <div id="CommentForm">
      <form action="" autoComplete="off" noValidate>
        <Input
          className="mr-2"
          multiline
          rowsMax={4}
          name="text"
          value={text}
          onChange={handleChange}
          placeholder="comment"
        />
        <Button
          className="ml-2"
          onClick={handleSubmit}
          variant="contained"
          size="small"
          color="primary"
        >
          Post
        </Button>
      </form>
    </div>
  );
};

export default CommentForm;
