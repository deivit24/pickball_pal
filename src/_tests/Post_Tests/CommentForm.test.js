import React from 'react';
import { render } from '@testing-library/react';
import CommentForm from '../../components/post/CommentForm';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing CommentForm

describe('Testing the CommentForm', () => {
  let postId = '0001';
  let color = 'white';
  it('it renders CommentForm without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <CommentForm postId={postId} color={color} />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <CommentForm postId={postId} color={color} />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
