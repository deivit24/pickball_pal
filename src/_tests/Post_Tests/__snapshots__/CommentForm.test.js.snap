// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the CommentForm matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="CommentForm"
  >
    <form
      action=""
      autocomplete="off"
      novalidate=""
    >
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline mr-2 white MuiInputBase-multiline MuiInput-multiline"
      >
        <textarea
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
          name="text"
          placeholder="comment"
          rows="1"
          style="height: 0px; overflow: hidden;"
        />
        <textarea
          aria-hidden="true"
          class="MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline"
          readonly=""
          style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); width: 100%;"
          tabindex="-1"
        />
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained ml-2 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Comment
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</DocumentFragment>
`;
